src/librustc_mir/interpret/visitor.rs

```diff
@@ -286,12 +289,24 @@ macro_rules! make_value_visitor {
                     layout::FieldPlacement::Arbitrary { ref offsets, .. } => {
                         // FIXME: We collect in a vec because otherwise there are lifetime errors:
                         // Projecting to a field needs (mutable!) access to `ecx`.
-                        let fields: Vec<EvalResult<'tcx, Self::V>> =
-                            (0..offsets.len()).map(|i| {
+                        let fields = if Self::WANT_FIELDS_SORTED {
```

Comment:
looks to me like we want a trait method on the machine for sorting instead of a bool that we check in the miri engine?

Reply:
This is a per-visitor decision, not a per-machine decision.

Reply:
ok, then a method on the visitor maybe?